src/services/filter/filters/midnight/helpers.rs

```diff
@@ -3,3 +3,443 @@
+
+use midnight_ledger::{
+	base_crypto::hash::{HashOutput, PERSISTENT_HASH_BYTES},
+	serialize::deserialize,
+	storage::DefaultDB,
+	structure::{Proof, Proofish, Transaction as MidnightNodeTransaction, TransactionHash},
+	transient_crypto::encryption,
+	zswap::{
+		keys::{SecretKeys, Seed},
+		CoinCiphertext,
+	},
+};
+use midnight_node_ledger_helpers::{CoinInfo, NetworkId, DB};
+
+/// Parse a transaction index item
+pub fn parse_tx_index_item<P: Proofish<DefaultDB>>(
+	hash_without_prefix: &str,
+	raw_tx_data: &str,
+	network_id: NetworkId,
+) -> Result<
+	(
+		TransactionHash,
+		Option<MidnightNodeTransaction<P, DefaultDB>>,
+	),
+	anyhow::Error,
+> {
+	let hash = hex::decode(hash_without_prefix)
+		.map_err(|e| anyhow::anyhow!("TransactionHashDecodeError: {}", e))?;
+
+	// TODO: For now we always return early because there is an issue with deserialising tx data from raw_tx_data while the Midnight node team investigates
+	// Remove this once the issue is fixed
+	// TransactionDeserializeError: Invalid input data for core::option::Option<midnight_ledger::structure::Transaction<midnight_ledger::structure::Proof, midnight_storage::db::InMemoryDB>>,
+	//                              received version: None, maximum supported version is None. Invalid discriminant: 4
+	if !raw_tx_data.is_empty() {
+		return Ok((
+			TransactionHash(HashOutput(
+				hash.try_into()
+					.map_err(|_| anyhow::anyhow!("Invalid hash length"))?,
+			)),
+			None,
+		));
+	}
+
+	// When testing, we don't have the raw tx data, so we just return the hash
+	if raw_tx_data.is_empty() {
+		return Ok((
+			TransactionHash(HashOutput(
+				hash.try_into()
+					.map_err(|_| anyhow::anyhow!("Invalid hash length"))?,
+			)),
+			None,
+		));
+	}
+
+	let (_hex_prefix, body_str) = raw_tx_data.split_at(2);
+	if hash.len() != PERSISTENT_HASH_BYTES {
+		return Err(anyhow::anyhow!(
+			"hash length ({}) != {PERSISTENT_HASH_BYTES}",
+			hash.len()
+		));
+	}
+	let hash = TransactionHash(HashOutput(
+		hash.try_into()
+			.map_err(|_| anyhow::anyhow!("Invalid hash length"))?,
+	));
+
+	// NOTE: Alternative way to decode the transaction if we want the 35 byte addresses as opposed to the 32 byte addresses
+	// 		 This method uses api.serialize() to serialize the addresses
+	// let tx = midnight_node_ledger::host_api::ledger_bridge::get_decoded_transaction(
+	// 	&[network_id as u8],
+	// 	body_str.as_bytes(),
+	// );
+
+	let body =
+		hex::decode(body_str).map_err(|e| anyhow::anyhow!("TransactionBodyDecodeError: {}", e))?;
+
+	let tx = deserialize(body.as_slice(), network_id)
+		.map_err(|e| anyhow::anyhow!("TransactionDeserializeError: {}", e))?;
+
+	Ok((hash, tx))
+}
+
+/// Map a MidnightChainType to a NetworkId
+pub fn map_chain_type(chain_type: &MidnightChainType) -> NetworkId {
+	match chain_type {
+		MidnightChainType::Development => NetworkId::TestNet,
+		MidnightChainType::Live => NetworkId::TestNet, // TODO: Change to MainNet once testnet-02 `system_chainType` returns `Development`
```

Comment:
This seems to be a bug on the Midnight RPC. For their testnet, it returns `Live` instead of `Development`. We may consider an alternative approach here by using `system_chain` and filtering for the word "testnet" instead in the future. For now this is OK since they don't have a `mainnet` yet.